src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py

```diff
@@ -206,1025 +201,78 @@ def create_dir(directory):
-            del attrs["cloudcoverpercentage"]
-
-        for key, val in attrs.items():
-            field = ogr.FieldDefn(key, val)
-            layer.CreateField(field)
-
-        # features
-        for idx in range(len(self._products_df_sorted["uuid"])):
-            wkt = self._products_df_sorted["footprint"].iloc[idx]
-            feature = ogr.Feature(layer.GetLayerDefn())
-            newgeom = ogr.CreateGeometryFromWkt(wkt)
-            # convert polygons to multi-polygons
-            newgeomtype = ogr.GT_Flatten(newgeom.GetGeometryType())
-            if newgeomtype == ogr.wkbPolygon:
-                multigeom = ogr.Geometry(ogr.wkbMultiPolygon)
-                multigeom.AddGeometryDirectly(newgeom)
-                feature.SetGeometry(multigeom)
-            else:
-                feature.SetGeometry(newgeom)
-            for key in attrs.keys():
-                if key == "ingestiondate":
-                    value = (
-                        self._products_df_sorted[key]
-                        .iloc[idx]
-                        .strftime("%Y-%m-%dT%H:%M:%SZ")
-                    )
-                else:
-                    value = self._products_df_sorted[key].iloc[idx]
-                feature.SetField(key, value)
-            layer.CreateFeature(feature)
-            feature = None
-
-        data_source = None
-
-        # coordinates of footprints are in WKT -> fp precision issues
-        # -> snap
-        gs.run_command(
-            "v.import",
-            input=tmp_name,
-            output=map_name,
-            layer=map_name,
-            snap=1e-10,
-            quiet=True,
-        )
-
-    def get_products_from_uuid_usgs(self, uuid_list):
-        scenes = []
-        for uuid in uuid_list:
-            metadata = self._api.metadata(uuid, "SENTINEL_2A")
-            scenes.append(metadata)
-        scenes_df = pandas.DataFrame.from_dict(scenes)
-        self._products_df_sorted = scenes_df
-        gs.message(
-            _("{} Sentinel product(s) found").format(len(self._products_df_sorted))
-        )
-
-    def set_uuid(self, uuid_list):
-        """Set products by uuid.
-
-        TODO: Find better implementation
-
-        :param uuid: uuid to download
-        """
-        if self._apiname == "USGS_EE":
-            self.get_products_from_uuid_usgs(uuid_list)
-        else:
-            from sentinelsat.sentinel import SentinelAPIError
+    normalized_datetime = datetime.fromisoformat(datetime_str)
+    if normalized_datetime.tzinfo is None:
+        return normalized_datetime.strftime("%Y-%m-%dT%H:%M:%S")
+    return normalized_datetime.astimezone(timezone.utc).strftime("%Y-%m-%dT%H:%M:%S")
+
+
+EODAG_PRODUCTTYPE_MAP = {
+    "S2MSI1C": "S2_MSI_L1C",
+    "S2MSI2A": "S2_MSI_L2A",
+    # Only found in wekeo, is S2MSI2Ap needed anymore?
+    # https://sentiwiki.copernicus.eu/web/s2-products#S2Products-XMLSchemaDefinitions(XSD)S2-Products-XML-Schema-Definitionstru
+    "S2MSI2Ap": "S2_MSI_L2AP",
```

Comment:
This was a pilot L2 product IIRC, @neteler do you think we can safely remove?

Reply:
Yes, IMHO no need to keep pilot products.